evalml/data_checks/null_data_check.py

```diff
@@ -0,0 +1,354 @@
+            ...             }
+            ...         ]
+            ...     },
+            ...     {
+            ...         "message": "Column(s) 'lots_of_null', 'few_null' have null values",
+            ...         "data_check_name": "NullDataCheck",
+            ...         "level": "warning",
+            ...         "details": {"columns": ["lots_of_null", "few_null"], "rows": None},
+            ...         "code": "COLS_WITH_NULL",
+            ...         "action_options": [
+            ...             {
+            ...                "code": "IMPUTE_COL",
+            ...                "data_check_name": "NullDataCheck",
+            ...                "metadata": {"columns": ["lots_of_null", "few_null"], "rows": None, "is_target": False},
+            ...                "parameters": {
+            ...                    "impute_strategies": {
+            ...                        "parameter_type": "column",
+            ...                        "columns": {
+            ...                            "lots_of_null": {"impute_strategy": {"categories": ["mean", "most_frequent"], "type": "category", "default_value": "mean"}},
+            ...                            "few_null": {"impute_strategy": {"categories": ["mean", "most_frequent"], "type": "category", "default_value": "mean"}}
+            ...                        }
+            ...                    }
+            ...                }
+            ...             }
+            ...         ]
+            ...     }
+            ... ]
+
+        """
+        messages = []
+
+        X = infer_feature_types(X)
+
+        highly_null_rows = NullDataCheck.get_null_row_information(
+            X, pct_null_row_threshold=self.pct_null_row_threshold
+        )
+        if len(highly_null_rows) > 0:
+            warning_msg = f"{len(highly_null_rows)} out of {len(X)} rows are {self.pct_null_row_threshold*100}% or more null"
+            rows_to_drop = highly_null_rows.index.tolist()
+
+            messages.append(
+                DataCheckWarning(
+                    message=warning_msg,
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.HIGHLY_NULL_ROWS,
+                    details={
+                        "rows": highly_null_rows.index.tolist(),
+                        "pct_null_cols": highly_null_rows,
+                    },
+                    action_options=[
+                        DataCheckActionOption(
+                            DataCheckActionCode.DROP_ROWS,
+                            data_check_name=self.name,
+                            metadata={"rows": rows_to_drop},
+                        )
+                    ],
+                ).to_dict()
+            )
+
+        highly_null_cols, _ = NullDataCheck.get_null_column_information(
+            X, pct_null_col_threshold=self.pct_null_col_threshold
+        )
+
+        X_to_check_for_any_null = X.ww.select(
```

Comment:
I think we can improve this implementation. Right now we do two scans of the data to determine the highly null columns and the columns with any nulls. Doing so also involves creating a new copy of the data without NaturalLanguage/DateTime. 

I think if we refactor `get_null_column_information` to return {col_name: % nulls} we only have to do one scan without creating another copy to determine which are highly null and which are null. 

Let's file an issue to do that in a separate PR.

Reply:
You're right, thank you for pointing out that nuance! I filed #3273 to track this :)